gui/src/2D/controls/toggleButton.ts

```diff
@@ -0,0 +1,308 @@
+import { Nullable } from "babylonjs/types";
+import { Observable } from "babylonjs/Misc/observable";
+import { Vector2 } from "babylonjs/Maths/math.vector";
+
+import { Rectangle } from "./rectangle";
+import { Control } from "./control";
+import { TextBlock } from "./textBlock";
+import { Image } from "./image";
+import { _TypeStore } from "babylonjs/Misc/typeStore";
+import { PointerInfoBase } from "babylonjs/Events/pointerEvents";
+
+/**
+ * Class used to create toggle buttons
+ */
+export class ToggleButton extends Rectangle {
+    /**
+     * Function called to generate the toActive animation
+     */
+    public toActiveAnimation: () => void;
+
+    /**
+     * Function called to generate the toInactive animation
+     */
+    public toInactiveAnimation: () => void;
+
+    /**
+     * Function called to generate a pointer enter animation when the toggle is active.
+     */
+    public pointerEnterActiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer out animation when the toggle is active.
+     */
+    public pointerOutActiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer down animation when the toggle is active.
+     */
+    public pointerDownActiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer up animation when the toggle is active.
+     */
+    public pointerUpActiveAnimation: () => void;
+
+    /**
+     * Function called to generate a pointer enter animation when the toggle is inactive.
+     */
+    public pointerEnterInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer out animation when the toggle is inactive.
+     */
+    public pointerOutInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer down animation when the toggle is inactive.
+     */
+    public pointerDownInactiveAnimation: () => void;
+    /**
+     * Function called to generate a pointer up animation when the toggle is inactive.
+     */
+    public pointerUpInactiveAnimation: () => void;
+
+    /** Observable raised when isActive is changed */
+    public onIsActiveChangedObservable = new Observable<boolean>();
+
+    /**
+     * Gets or sets a boolean indicating that the toggle will let internal controls handle picking instead of doing it directly using its bounding info
+     */
+    public delegatePickingToChildren = false;
+
+    private _image: Nullable<Image>;
+    /**
+     * Returns the image part of the button (if any)
+     */
+    public get image(): Nullable<Image> {
+        return this._image;
+    }
+
+    private _textBlock: Nullable<TextBlock>;
+    /**
+     * Returns the image part of the button (if any)
+     */
```

Comment:
Wrong doc (copy/paste of the previous one)

Reply:
This should be resolved in the latest update to this pull request.